be/src/column/field.cpp

```diff
@@ -44,4 +44,23 @@ ColumnPtr Field::create_column() const {
     return ChunkHelper::column_from_field(*this);
 }
 
+FieldPtr Field::convert_to_dict_field(const Field& field) {
+    if (field.type()->type() == TYPE_VARCHAR) {
+        FieldPtr res = std::make_shared<Field>(field);
+        res->_type = get_type_info(TYPE_INT);
+        return res;
+    } else if (field.type()->type() == TYPE_ARRAY && field.sub_field(0).type()->type() == TYPE_VARCHAR) {
```

Comment:
don't support nested string array? like array<array<string>>

Reply:
can it support TYPE_BINARY?

Reply:
don't support now